src/legacy/core_plugins/kibana/public/visualize/np_ready/listing/visualize_listing.js

```diff
@@ -26,30 +26,38 @@ import { i18n } from '@kbn/i18n';
 import { getServices } from '../../kibana_services';
 import { wrapInI18nContext } from '../../legacy_imports';
 
+import { syncQueryStateWithUrl } from '../../../../../../../plugins/data/public';
+
 export function initListingDirective(app) {
   app.directive('visualizeListingTable', reactDirective =>
     reactDirective(wrapInI18nContext(VisualizeListingTable))
   );
 }
 
-export function VisualizeListingController($injector, $scope, createNewVis) {
+export function VisualizeListingController($injector, $scope, createNewVis, kbnUrlStateStorage) {
   const {
     addBasePath,
     chrome,
     savedObjectsClient,
     savedVisualizations,
-    data: {
-      query: {
-        timefilter: { timefilter },
-      },
-    },
+    data: { query },
     toastNotifications,
     uiSettings,
     visualizations,
     core: { docLinks, savedObjects },
   } = getServices();
   const kbnUrl = $injector.get('kbnUrl');
 
+  // syncs `_g` portion of url with query services
+  const { stop: stopSyncingQueryServiceStateWithUrl } = syncQueryStateWithUrl(
```

Comment:
The thing I wanted to discuss also,
should we sync the global state with the query on the listing page?
Seems it's redundant (at least if you don't want to share a link to a listing page with the global state)...

Reply:
I think we do want to sync it, otherwise in the following scenario you would lose your global state:
* Go to a visualization
* Change global state (add pinned filter, change time range)
* Go to listing page
* Reload
* Click visualization
* Global state is reset

Reply:
Makes sense!
I suppose we should set up a `test-it-week`,
while this knowledge is still fresh. 😃